src/pool.rs

```diff
@@ -114,12 +114,14 @@ impl ConnectionPool {
 
                         let address = Address {
                             id: address_id,
-                            database: pool_name.clone(),
+                            database: shard.database.clone(),
```

Comment:
I think this was an oversight in the initial multi-db PR. `database` should point to the actual database name and not the pool name